pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -273,11 +278,29 @@ public RelDataType toType(RelDataTypeFactory typeFactory) {
       }
     },
     TIMESTAMP(LONG, NullValuePlaceHolder.LONG) {
+      @Override
+      public RelDataType toType(RelDataTypeFactory typeFactory) {
+        return typeFactory.createSqlType(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE);
+      }
+    },
+    TIMESTAMP_NTZ(LONG, NullValuePlaceHolder.LONG) {
       @Override
       public RelDataType toType(RelDataTypeFactory typeFactory) {
         return typeFactory.createSqlType(SqlTypeName.TIMESTAMP);
       }
     },
```

Comment:
I guess you did this to follow the Spark terminology, but apart from my preference to following Postgres terminology I think this is breaking backward compatibility. Before this PR a Pinot TIMESTAMP was converted to Calcite TIMESTAMP and now we are converting it into a TIMESTAMP WITH LOCAL TIME ZONE.

I may be wrong but I think what we should be doing here is to return that our TIMESTAMP_TZ returns a Calcite TIMESTAMP_TZ